app/templates/components/forms/wizard/sponsors-step.hbs

```diff
@@ -20,51 +20,53 @@
 
   {{#if data.event.isSponsorsEnabled}}
     {{#each data.event.sponsors as |sponsor index|}}
-      <h4>
-        {{t 'Sponsor'}} #{{inc index}}
-        {{#if data.event.sponsors.length}}
-          <button type="button" class="ui mini red button right floated" {{action 'removeSponsor' sponsor}}>
-            <i class="trash icon"></i>
-            {{t 'Remove'}}
-          </button>
-        {{/if}}
-      </h4>
-      <br>
-      <div class="fields">
-        <div class="five wide field">
-          <label class="required">{{t 'Name'}}</label>
-          {{input type='text' value=sponsor.name name='name'}}
-        </div>
-        <div class="two wide field">
-          <label>{{t 'Level'}}</label>
-          {{input type='number' value=sponsor.level name='level' min=0}}
-        </div>
-        <div class="three wide field">
-          <label>{{t 'Type'}}</label>
-          {{input type='text' value=sponsor.type name='type'}}
-        </div>
-        <div class="six wide field">
-          <label>{{t 'URL'}}</label>
-          {{widgets/forms/link-input segmentedLink=sponsor.segmentedLink name='url'}}
-        </div>
-      </div>
-      <div class="fields">
-        <div class="five wide computer eight wide tablet field">
-          <label>{{t 'Description'}}</label>
-          {{widgets/forms/rich-text-editor value=sponsor.description name='description'}}
+      {{#unless sponsor.isDeleted}}
```

Comment:
Not the right way to do it (for example, for the index, the delete record will also be taken into account). Instead the array you loop over should be filtered out. Use a computed property and loop over that.